src/main/src/main/java/org/geoserver/system/status/OSHISystemInfoCollector.java

```diff
@@ -104,433 +104,439 @@ public OSHISystemInfoCollector() {
     @Override
     List<MetricValue> retrieveSystemInfo(MetricInfo info) {
         List<MetricValue> si = super.retrieveSystemInfo(info);
-        try {
-            switch (info) {
-                    // system metrics
-                case OPERATING_SYSTEM:
-                    {
-                        MetricValue mv = new MetricValue(info);
-                        mv.setAvailable(true);
-                        mv.setValue(os.getFamily() + " " + os.getVersionInfo().getVersion());
-                        si = Collections.singletonList(mv);
-                        break;
-                    }
-                case SYSTEM_UPTIME:
-                    {
-                        MetricValue mv = new MetricValue(info);
-                        mv.setAvailable(true);
-                        mv.setValue(os.getSystemUptime());
-                        si = Collections.singletonList(mv);
-                        break;
-                    }
-                case SYSTEM_AVERAGE_LOAD_1:
-                    {
-                        MetricValue mv = new MetricValue(info);
-                        double lv = pr.getSystemLoadAverage(1)[0];
-                        if (lv > 0) {
+        if (super.getStatisticsStatus()) {
```

Comment:
If I read correclty, the is basically wraps the entire method body... if so, please invert the logic and exit right away, instead of nesting code. 
There is a lot of articles on the subject on the internet, try to search for "avoid deeply nested code". For a quick reference about this particular case, see "early return" [here](https://en.wikibooks.org/wiki/Computer_Programming/Coding_Style/Minimize_nesting#:~:text=Other%20than%20refactoring%20or%20avoiding,and%20cleanup%20are%20both%20visible)).

Reply:
And oh, BTW, this code is NPE prone, getStatitsticsStatus() may return null. Please also add a test covering the case of the setting being missing, which will happen on upgrade (alternative, use a Boolean for the field, but return boolean from the getter, considering nulls too, e.g., ``return Boolean.TRUE.equals(statisticsEnabled))``

Reply:
Applied the feedback!